airflow-core/src/airflow/api_fastapi/core_api/routes/public/assets.py

```diff
@@ -367,8 +373,8 @@ def materialize_asset(
             f"More than one DAG materializes asset with ID: {asset_id}",
         )
 
-    dag: DAG | None
-    if not (dag := dag_bag.get_dag(dag_id)):
+    dag: DAG | None = get_dag_from_dag_bag(dag_bag, dag_id)
```

Comment:
Typing is already infered from return type annotation of the function.
```suggestion
    dag = get_dag_from_dag_bag(dag_bag, dag_id)
```